petclinic.yaml

```diff
@@ -11,16 +11,21 @@ tools:
     type: dockerImage
     image: maven:3.5.4-jdk-8
     entryPoint: sleep infinity
-    persistentVolumes:
+    volumes:
       - name: maven-repo
-        containerPath: /root/.m2
-    services:
+        mountPath: /root/.m2
+      - name: projects
+        mountPath: /projects
+    endpoints:
       - name: spring-boot
-        port: 8080
-    mountProjectsSources: true
```

Comment:
I understand that you remove that because you have moved it to the volumes section. But the problem with this approach is that the user need to know Che internals (i.e. that Che does the cloning in `/projects/`) and ideally I would like to avoid that.
So while I am perfectly fine if we don't implement `mountProjectSources` in Phase 1 I would want us to implement it in the future.

Reply:
>  (i.e. that Che does the cloning in /projects/)

It doesn't clone projects to `/projects/` but it clones projects to volume with name `projects`. So, a user is able to specify any mountPath he wishes.

I removed `mountProjectSources` to have the same format as `che-plugin` uses. We can consider adding `mountProjectSources` to both of Devfile and che-plugin. Or you think that it is more critical for Devfile but it is not an issue for `che-plugin.yaml`?

Reply:
> it clones projects to volume with name projects. So, a user is able to specify any mountPath he wishes.

I am probably a n00b Che user but I wasn't aware of this convention. And we should make things easier to understand for n00bs like me.

I agree with you that `che_plugin.yaml` and `devfile` should share the same spec so +1 to update/review both models. I think this example has helped us to identify some problems with che_plugin.yaml and that's an opportunity to improve it. I have created https://github.com/eclipse/che/issues/12249 to track that.